### PR TITLE
REL-1198: List value URI definition should work with neo4j RDF storage.

### PR DESCRIPTION
**Pull request by vbyndych**
## Goal
List value URIs can be modified via user interface. Yet implementation is modifying SQL statements table directly, and would not work with neo4j RDF storage.

## Changelog
- feat: extended complex search functionality to fetch lists data.
- chore: removed unused constructors and extra spaces.
- refactor: changed logic for values fetching using new complex search tripple method for consistency.

## Dependencies
https://github.com/oat-sa/lib-generis-search/pull/32

### CODE REVIEW

**Review by yaraslau-kavaliou — APPROVED** [review]
Great job!
Tested locally with sql and neo4j databases and updating uri functionality for lists works without issues.

- [x]  New code respects code style rules
- [x]  New code respects best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful
- [x]  Pull request's target is not master
- [x]  Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x]  Commits messages are meaningful
- [x]  Commits are atomic

**Review by augustas — APPROVED** [review]
Looking good and working good for me

- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

File: core/kernel/persistence/interface.ClassInterface.php [hunk 1]
```diff
@@ -222,4 +222,14 @@ public function createInstanceWithProperties(core_kernel_classes_Class $type, $p
      * @return boolean
      */
     public function deleteInstances(core_kernel_classes_Class $resource, $resources, $deleteReference = false);
+
+    /**
+     * Changes class URI for all its properties and linked objects.
+     *
+     * @param core_kernel_classes_Class $resource
```
[the remark this answers is not in the document]
Comment (vbyndych) [reply core/kernel/persistence/interface.ClassInterface.php:229]: I totally agree. Will fix.

File: core/kernel/persistence/starsql/search/GateWay.php [hunk 2]
```diff
@@ -76,13 +77,55 @@ public function connect()
...
+     * @param bool $isDistinct
+     *
+     * @return ResultSetInterface
+     */
+    public function searchTriples(QueryBuilderInterface $Builder, string $propertyUri, bool $isDistinct = false)
+    {
+        $result = $this->fetchTripleList(
+            parent::searchTriples($Builder, $propertyUri, $isDistinct)
+        );
+        return new $this->resultSetClassName($result, count($result));
```
[the remark this answers is not in the document]
Comment (vbyndych) [reply core/kernel/persistence/starsql/search/GateWay.php:98]: will fix

**Pull request merged**